### PR TITLE
implement order workflow

### PR DESCRIPTION
**Pull request by tonhuynh**
- add two new fields: workflowId and workflowStatus
- override CustomOrder by CustomOrderWorkflow

JIRA: VT-25

### CODE REVIEW

File: VirtoCommerce.OrderModule.Data/Services/CustomerOrderBuilderImpl.cs [hunk 1]
```diff
@@ -31,7 +32,7 @@ public virtual orderModel.CustomerOrder PlaceCustomerOrderFromCart(cartModel.Sho
         #endregion
         protected virtual orderModel.CustomerOrder ConvertCartToOrder(cartModel.ShoppingCart cart)
         {
-            var retVal = AbstractTypeFactory<orderModel.CustomerOrder>.TryCreateInstance();
+            var retVal = AbstractTypeFactory<CustomerOrderWorkflow>.TryCreateInstance();
```
Comment (eMazeika) [inline VirtoCommerce.OrderModule.Data/Services/CustomerOrderBuilderImpl.cs]: don't change this line. Do the type casting later, when you really need to have CustomerOrderWorkflow

File: VirtoCommerce.OrderModule.Web/Module.cs [hunk 2]
```diff
@@ -170,6 +172,11 @@ public override void PostInitialize()
             var allOrderKnownTypes = new[] { typeof(Shipment), typeof(PaymentIn), typeof(CustomerOrder) }.Concat(allPaymentMethodTypes).Concat(allShippingmethodTypes).ToArray();
             httpConfiguration.Formatters.XmlFormatter.SetSerializer<CustomerOrder>(new DataContractSerializer(typeof(CustomerOrder), allOrderKnownTypes));
             httpConfiguration.Formatters.XmlFormatter.SetSerializer<CustomerOrderSearchResult>(new DataContractSerializer(typeof(CustomerOrderSearchResult), allOrderKnownTypes));
+
+            //customer order workflow
+            AbstractTypeFactory<IOperation>.OverrideType<CustomerOrder, CustomerOrderWorkflow>();
```
Comment (eMazeika) [inline VirtoCommerce.OrderModule.Web/Module.cs]: is this line really needed? Can remove it?

File: VirtoCommerce.OrderModule.Web/Module.cs [hunk 3]
```diff
@@ -170,6 +172,11 @@ public override void PostInitialize()
...
+            AbstractTypeFactory<CustomerOrder>.OverrideType<CustomerOrder, CustomerOrderWorkflow>()
+                .WithFactory(() => new CustomerOrderWorkflow { OperationType = "CustomerOrder" });
```
Comment (eMazeika) [inline VirtoCommerce.OrderModule.Web/Module.cs]: remove the factory method. Assign the OperationType inside CustomerOrderWorkflow constructor

**Review by eMazeika — CHANGES_REQUESTED** [review]
3 comments added.
But where are changes to persistency layer, migrations, etc?

File: VirtoCommerce.OrderModule.Web/Module.cs (hunk 3 above)
Comment (tonhuynh) [reply VirtoCommerce.OrderModule.Web/Module.cs]: it's fixed

File: VirtoCommerce.OrderModule.Web/Module.cs (hunk 2 above)
Comment (tonhuynh) [reply VirtoCommerce.OrderModule.Web/Module.cs]: Yes, It was removed

File: VirtoCommerce.OrderModule.Data/Services/CustomerOrderBuilderImpl.cs (hunk 1 above)
Comment (tonhuynh) [reply VirtoCommerce.OrderModule.Data/Services/CustomerOrderBuilderImpl.cs]: it's fixed

**Review by eMazeika — APPROVED** [review]
ok

**Pull request merged**